sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -84,8 +99,24 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   private async getToken(options: GetTokenOptions): Promise<string | undefined> {
     let accessToken = this.tokenCache.getCachedToken();
     if (accessToken === undefined) {
-      accessToken = (await this.credential.getToken(this.scopes, options)) || undefined;
-      this.tokenCache.setCachedToken(accessToken);
+      // Waiting for the next refresh (or forcefully triggering a new refresh)
+      // only if the cache is unable to retrieve the access token,
+      // which means that it has expired, or it has never been set.
+      const refreshPromise = this.tokenRefresher.forcedRefresh(options);
+      if (refreshPromise !== null) {
+        accessToken = await refreshPromise;
+      }
+    } else {
+      // If we still have a cached access token,
+      // then attempt to refresh without waiting.
+      const refreshPromise = this.tokenRefresher.refresh(options);
```

Comment:
Why refresh if the token cache's refresh time hasn't been reached?  We use a [delta](https://github.com/azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/credentials/accessTokenCache.ts#L9) to determine the time before the real expiration time that we should start a refresh.  The previous logic case (`accessToken === undefined`) should handle that, in theory.

Reply:
This has changed, but the gist is that we will not attempt to refresh based on a conditional. We can make the conditional better too! if that makes sense.